### PR TITLE
Tokenize input properly

### PR DESCRIPTION
**Pull request by themattchan**
Putting this up for preliminary comments/review. I will test this tomorrow when I get `nix` working again after a mac os upgrade.

The major change is to remove the comment-stripping hack and all the extraneous calls to `whiteSpace` (whitespace should be handled automatically by parsers using the `TokenParsing` class).

### CODE REVIEW

**Comment by intractable** [comment]
Oh, this is nice. Thanks!

Just FYI the plan is to (eventually) drop our custom lexing/parsing altogether, in order to play nicely as a `protoc` plugin (i.e, we'll get the AST from that tool instead), so I wouldn't push too terribly hard on those components unless they're blocking you.

That said, using `protoc` will be a bit work and likely won't be done soon, so fixing things up like this great in the interim.

**Comment by themattchan** [comment]
This works and passes tests on my machine now, and is ready for review.

File: src/Proto3/Suite/DotProto/Parsing.hs [hunk 1]
```diff
@@ -45,75 +48,45 @@ parseProtoFile modulePath =
...
--- this much at all in the very near short term, let's just pay the freight and
--- use `Text.Parsec.Token` or somesuch.
-data StripCommentState
-       --               | starts  | ends | error on?       |
-  = BC -- block comment | "/*"    | "*/" | mismatch        |
-  | DQ -- double quote  | '"'     | '"'  | mismatch        |
-  | LC -- line comment  | "//"    | '\n' | mismatch ('\n') |
-  deriving Show
+empty :: ProtoParser ()
+empty = whiteSpace >> textSymbol ";" >> return ()
```
[the remark this answers is not in the document]
Comment (themattchan) [reply src/Proto3/Suite/DotProto/Parsing.hs]: 👍 let me try this...

**Comment by themattchan** [comment]
CI is broken, merging.

**Pull request merged**